glib2/ext/glib2/rbgobj_object.c

```diff
@@ -621,30 +621,112 @@ rg_type_name(VALUE self)
+                                         GValue *to_value, gpointer user_data)
+{
+    VALUE rb_from_value = rbgobj_gvalue_to_rvalue(from_value);
+    VALUE rb_to_value = rbgobj_gvalue_to_rvalue(to_value);
+    RGBindPropertyCallbackData *data = (RGBindPropertyCallbackData *)user_data;
+    VALUE proc = data->transform_from_callback;
+    rb_to_value = rb_funcall(proc, rb_intern("call"), 1, rb_from_value);
+    rbgobj_rvalue_to_gvalue(rb_to_value, to_value);
+    return TRUE;
+}
+
+static void
+rg_destroy_bind_property_full_data(gpointer data)
+{
+    free((void *) data);
+}
+
 static VALUE
-rg_bind_property(VALUE self,
-                 VALUE rb_source_property,
-                 VALUE rb_target,
-                 VALUE rb_target_property,
-                 VALUE rb_flags)
+rg_bind_property(gint argc, VALUE *argv, VALUE self)
 {
+    VALUE rb_source_property;
+    VALUE rb_target;
+    VALUE rb_target_property;
+    VALUE rb_flags;
+    VALUE rb_options;
+    VALUE rb_transform_to;
+    VALUE rb_transform_from;
+
     gpointer source;
     const gchar *source_property;
     gpointer target;
     const gchar *target_property;
     GBindingFlags flags;
     GBinding *binding;
+    GBindingTransformFunc transform_to;
+    GBindingTransformFunc transform_from;
+
+    rb_scan_args(argc, argv, "41", &rb_source_property, &rb_target,
+                 &rb_target_property, &rb_flags, &rb_options);
+
+    rbg_scan_options(rb_options,
+                     "transform_to", &rb_transform_to,
+                     "transform_from", &rb_transform_from,
+                     NULL);
 
     source = RVAL2GOBJ(self);
     source_property = RVAL2CSTR(rb_source_property);
     target = RVAL2GOBJ(rb_target);
     target_property = RVAL2CSTR(rb_target_property);
     flags = RVAL2GBINDINGFLAGS(rb_flags);
 
-    binding = g_object_bind_property(source, source_property,
-                                     target, target_property,
-                                     flags);
+    if (argc == 4) {
+        binding = g_object_bind_property(source, source_property,
+                                         target, target_property,
+                                         flags);
+    } else {
+        if(NIL_P(rb_transform_to))
+            transform_to = NULL;
+        else
+            transform_to = rg_bind_property_transform_to_callback;
+
+        if(NIL_P(rb_transform_from))
+            transform_from = NULL;
+        else
+            transform_from = rg_bind_property_transform_from_callback;
+
+        if (!transform_to && !transform_from) {
+            binding = g_object_bind_property(source, source_property,
+                                             target, target_property,
+                                             flags);
+        } else {
+            RGBindPropertyCallbackData *data;
+            data = (RGBindPropertyCallbackData *)malloc(sizeof(RGBindPropertyCallbackData));
```

Comment:
> xmalloc() cares Ruby's GC.

Searching in internet, I found that : 
*  xmalloc allocates memory from within a region of memory that Ruby itself has allocated with libc malloc
*  The functions Ruby_xmalloc() and Ruby_xrealloc() are used by a variety of internal functions to process requests for additional memory.   When a memory allocation request fails,  these functions call garbage_collect()  to free up memory.   A second failure will cause Ruby to generate a memory allocation error and exit.

Is it what you wanted to say or is there more (those informations I found where on old sites)?